Lib/idlelib/idle_test/test_query.py

```diff
@@ -242,6 +242,57 @@ def test_entry_ok_helpsource(self):
                 self.assertEqual(dialog.entry_ok(), result)
 
 
+class CustomRunCLIargsokTest(unittest.TestCase):
+    "Test cli_ok method of the CustomRun subclass of Query."
+
+    class Dummy_CustomRun:
+        cli_args_ok = query.CustomRun.cli_args_ok  # Function being tested.
+        entry = Var()
+        entry_error = {}
+        def __init__(self, dummy_entry):
+            self.entry.set(dummy_entry)
+            self.entry_error['text'] = ''
+        def showerror(self, message):
+            self.entry_error['text'] = message
+
+    def test_blank_args(self):
+        dialog = self.Dummy_CustomRun(' ')
+        self.assertEqual(dialog.cli_args_ok(), None)
+        self.assertIn('no arguments', dialog.entry_error['text'])
+
+    def test_invalid_args(self):
+        dialog = self.Dummy_CustomRun("'no-closing-quote")
+        self.assertEqual(dialog.cli_args_ok(), None)
+        self.assertIn('No closing', dialog.entry_error['text'])
```

Comment:
This should use assertRegex to do a case-insensitive check.
```suggestion
        self.assertRegex(r'(?i)no closing', dialog.entry_error['text'])
```

Reply:
The test is clear and it works.